lib/ObjWriter/objwriter.cpp

```diff
@@ -550,22 +547,119 @@ extern "C" void EmitDebugFileInfo(ObjectWriter *OW, int FileId,
   OST.EmitCVFileDirective(FileId, FileName);
 }
 
-// This should be invoked at the end of function code emission to flush
-// debug function info.
-extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
-                                      const char *FunctionName,
-                                      int FunctionSize) {
+static void EmitPDBDebugVarInfo(MCObjectStreamer &OST, const MCSymbol *Fn,
+                                DebugVarInfo LocInfos[], int NumVarInfos) {
+  MCContext &OutContext = OST.getContext();
+  const MCObjectFileInfo *MOFI = OutContext.getObjectFileInfo();
+  assert(MOFI->getObjectFileType() == MOFI->IsCOFF);
+
+  for (int i = 0; i < NumVarInfos; i++) {
+    // Emit an S_LOCAL record
+    DebugVarInfo var = LocInfos[i];
+    LocalSym sym;
+
+    memset(&sym, 0, sizeof(sym));
+
+    int sizeofSym = sizeof(LocalSym);
+    short rectyp = SymbolRecordKind::S_LOCAL;
+    short reclen = sizeof(rectyp) + sizeofSym + var.name.length() + 1;
```

Comment:
It's okay to set it explicitly but it gives us a hint what we are emitting.
With regard of sizeof(rectyp) which is always 2 because we don't count the length field (2byte) but starting with kind field. Such subtle meaning is defined in the header file at least.